### ISSUE TITLE
Waiting for PostgreSQL Server to start...

### ISSUE THREAD

**Issue by eldadh** [roflcoopter/viseron#748]
I get the fallowing error in the logs:

Starting PostgreSQL Server...
s6-applyuidgid: fatal: unable to exec /usr/lib/postgresql//bin/postgres: No such file or directory
/var/run/postgresql:5432 - no response
Waiting for PostgreSQL Server to start...

running version 3.0.0b4 on jetson nano 2gb
apparently there are two "\" backslashes in the command

the problem was there in the previous version as well b3
eldad

**Comment by roflcoopter** [comment]
Hmm seems that the postgresql version is not found correctly.
If you look furhter up in the logs, do you see a log line that says "PostgreSQL major version"?

**Comment by eldadh** [comment]
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 10-adduser: executing...
************************ UID/GID *************************
User uid:    0
User gid:    0
************************** Done **************************
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-gid-video-device: executing...
[cont-init.d] 20-gid-video-device: exited 0.
[cont-init.d] 30-edgetpu-permission: executing...
************** Setting EdgeTPU permissions ***************
Coral Vendor IDs:
"1a6e"
"18d1"
No EdgeTPU USB device was found
No EdgeTPU PCI device was found
************************** Done **************************
[cont-init.d] 30-edgetpu-permission: exited 0.
[cont-init.d] 40-set-env-vars: executing...
****** Checking for hardware acceleration platforms ******
OpenCL cannot be used
VA-API cannot be used
CUDA is available!
*********************** Done *****************************
[cont-init.d] 40-set-env-vars: exited 0.
[cont-init.d] 50-check-if-rpi: executing...
********** Checking if we are running on an RPi **********
Not running on any supported RPi
*********************** Done *****************************
[cont-init.d] 50-check-if-rpi: exited 0.
[cont-init.d] 55-check-if-jetson: executing...
****** Checking if we are running on a Jetson Board ******
Running on a Jetson Nano
*********************** Done *****************************
[cont-init.d] 55-check-if-jetson: exited 0.
[cont-init.d] 60-ffmpeg-path: executing...
****************** Getting FFmpeg path *******************
FFmpeg path: /usr/bin/ffmpeg
*********************** Done *****************************
[cont-init.d] 60-ffmpeg-path: exited 0.
[cont-init.d] 70-gstreamer-path: executing...
***************** Getting GStreamer path *****************
GStreamer path: /usr/bin/gst-launch-1.0
*********************** Done *****************************
[cont-init.d] 70-gstreamer-path: exited 0.
[cont-init.d] 80-postgres: executing...
***************** Preparing PostgreSQL *******************
Database has not been initialized. Initializing...
s6-applyuidgid: fatal: unable to exec /usr/lib/postgresql/12/bin/initdb: No such file or directory
Starting PostgreSQL...
s6-applyuidgid: fatal: unable to exec /usr/lib/postgresql/12/bin/pg_ctl: No such file or directory
psql: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
Database has not been created. Creating...
createdb: could not connect to database template1: could not connect to server: No such file or directory
        Is the server running locally and accepting
        connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
Stopping PostgreSQL...
s6-applyuidgid: fatal: unable to exec /usr/lib/postgresql/12/bin/pg_ctl: No such file or directory
*********************** Done *****************************
[cont-init.d] 80-postgres: exited 0.
[cont-init.d] done.
[services.d] starting services
Starting PostgreSQL Server...
s6-applyuidgid: fatal: unable to exec /usr/lib/postgresql/12/bin/postgres: No such file or directory
[services.d] done.
/var/run/postgresql:5432 - no response
Waiting for PostgreSQL Server to start...

**Comment by roflcoopter** [comment]
Fixed in v3.0.0b5, now uses another method of getting the PostgreSQL version which should work on the Nano container as well.
Feel free to re-open if it persists

**Comment by toonsevrin** [comment]
```
2024-05-22 09:25:36.373 UTC [521] FATAL:  database "viseron" does not exist
[2024-05-22 09:25:36] [ERROR   ] [viseron.components] - Uncaught exception setting up component storage: (psycopg2.OperationalError) connection to server at "localhost" (::1), port 5432 failed: FATAL:  database "viseron" does not exist
```

Just moved to viseron:3.0.0b6 from b3 and got this.
